OpenSim/Common/Component.cpp

```diff
@@ -202,19 +208,25 @@ void Component::connect(Component &root)
 
 
 // Call connect on all components and find unconnected Connectors a
-void Component::componentsConnect(Component& root) const
+void Component::componentsConnect(Component& root) 
 {
     // First give the subcomponents the opportunity to connect themselves
-    for(unsigned int i=0; i<_components.size(); i++){
-        _components[i]->connect(root);
+    for (unsigned int i = 0; i<_memberSubcomponents.size(); ++i) {
+        _memberSubcomponents[i].upd()->connect(root);
+    }
+    for(unsigned int i=0; i<_propertySubcomponents.size(); ++i){
+        _propertySubcomponents[i].get()->connect(root);
+    }
+    for (unsigned int i = 0; i<_adoptedSubcomponents.size(); ++i) {
+        _adoptedSubcomponents[i].upd()->connect(root);
     }
 }
 
 void Component::disconnect()
 {
     // First give the subcomponents the opportunity to disconnect themselves
-    for (unsigned int i = 0; i<_components.size(); i++){
-        _components[i]->disconnect();
+    for (unsigned int i = 0; i<_propertySubcomponents.size(); i++){
```

Comment:
@tkuchida Good question. The reason is that unlike the `_propertySubcomponents` the size of member and adopted are fixed (there is no unadopt/abondon subcomponent yet) so there is no concern that I will have a refrence to stale Component in a connector as could happen with a property component that could change or be removed. 

It wouldn't hurt disconnect them also since connect will be called and it disconnects them first before connecting.